actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -437,7 +437,14 @@ def handle_options(options)
           end
 
           options[:path]      ||= "/"
-          options[:same_site] ||= request.cookies_same_site_protection
+
+          same_site_protection = if request.cookies_same_site_protection.is_a?(Proc)
```

Comment:
```suggestion
          same_site_protection = if request.cookies_same_site_protection.respond_to?(:call)
```

Reply:
Wonder if we want to auto convert a passed non-proc value into a proc when assigning the config?

Reply:
We can do this.